pkg/cli/start.go

```diff
@@ -118,9 +119,53 @@ func initInsecure() error {
 	return nil
 }
 
+// maxSizePerProfile is the maximum total size in bytes for profiles per
+// profile type.
+var maxSizePerProfile = envutil.EnvOrDefaultInt64(
+	"COCKROACH_MAX_SIZE_PER_PROFILE", 100<<20 /* 100 MB */)
+
+// gcProfiles removes old profiles matching the specified prefix when the sum
+// of newer profiles is larger than maxSize. Requires that the suffix used for
+// the profiles indicates age (e.g. by using a date/timestamp suffix) such that
+// sorting the filenames corresponds to ordering the profiles from oldest to
+// newest.
+func gcProfiles(prefix string, maxSize int64) {
+	paths, err := filepath.Glob(prefix + "*")
```

Comment:
nit: ioutil.ReadDir will returns a sorted list of `os.FileInfo`, doing the sorting and the stat for you. You can then call `filepath.Match` on it to check the glob.